### PR TITLE
[SPARK-40675][DOCS] Supplement undocumented spark configurations in `configuration.md`

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR aims to supplement missing spark configurations in `org.apache.spark.internal.config` in `configuration.md`.

### Why are the changes needed?

Help users to confirm configuration through documentation instead of code.

### Does this PR introduce _any_ user-facing change?

Yes, more configurations in documentation.

### How was this patch tested?

Pass the GitHub Actions.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -1993,6 +1995,24 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.eventLog.gcMetrics.youngGenerationGarbageCollectors</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1999]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L195-L203

File: docs/configuration.md [hunk 2]
```diff
@@ -1993,6 +1995,24 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>Copy,PS Scavenge,ParNew,G1 Young Generation</td>
+  <td>
+    Names of supported young generation garbage collector. A name usually is the return of GarbageCollectorMXBean.getName.
+    The built-in young generation garbage collectors are Copy,PS Scavenge,ParNew,G1 Young Generation.
+  </td>
+  <td>3.0.0</td>
+</tr>
+<tr>
+  <td><code>spark.eventLog.gcMetrics.oldGenerationGarbageCollectors</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2008]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L205-L213

File: docs/configuration.md [hunk 3]
```diff
@@ -349,6 +349,23 @@ of the most common options to set are:
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.executor.allowSparkContext</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:353]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L2206-L2211

File: docs/configuration.md [hunk 4]
```diff
@@ -468,6 +485,43 @@ of the most common options to set are:
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.decommission.enabled</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:489]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L2133-L2144

File: docs/configuration.md [hunk 5]
```diff
@@ -468,6 +485,43 @@ of the most common options to set are:
...
+<tr>
+  <td><code>spark.executor.decommission.killInterval</code></td>
+  <td>(none)</td>
+  <td>
+    Duration after which a decommissioned executor will be killed forcefully by an outside (e.g. non-spark) service.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.executor.decommission.forceKillTimeout</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:509]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L2158-L2165

File: docs/configuration.md [hunk 6]
```diff
@@ -681,14 +735,24 @@ Apart from these, the following properties are also available, and may be useful
 </tr>
 <tr>
   <td><code>spark.redaction.regex</code></td>
-  <td>(?i)secret|password|token</td>
+  <td>(?i)secret|password|token|access[.]key</td>
```
Comment (dcoliversun) [inline docs/configuration.md:738]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1116-L1124

File: docs/configuration.md [hunk 7]
```diff
@@ -681,14 +735,24 @@ Apart from these, the following properties are also available, and may be useful
...
   <td>
     Regex to decide which Spark configuration properties and environment variables in driver and
     executor environments contain sensitive information. When this regex matches a property key or
     value, the value is redacted from the environment UI and various logs like YARN and event logs.
   </td>
   <td>2.1.2</td>
 </tr>
+<tr>
+  <td><code>spark.redaction.string.regex</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:747]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1126-L1133

File: docs/configuration.md [hunk 8]
```diff
@@ -906,6 +978,23 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>1.4.0</td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.unsafe.file.output.buffer</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:982]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1339-L1348

File: docs/configuration.md [hunk 9]
```diff
@@ -906,6 +978,23 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>32k</td>
+  <td>
+    The file system for this buffer size after each partition is written in unsafe shuffle writer. 
+    In KiB unless otherwise specified.
+  </td>
+  <td>2.3.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.spill.diskWriteBufferSize</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:991]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1350-L1358

File: docs/configuration.md [hunk 10]
```diff
@@ -1028,6 +1128,14 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>1.1.1</td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.sort.io.plugin.class</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1132]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1320-L1325

File: docs/configuration.md [hunk 11]
```diff
@@ -1063,6 +1171,58 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>2.3.0</td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.reduceLocality.enabled</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1175]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1534-L1539

File: docs/configuration.md [hunk 12]
```diff
@@ -1063,6 +1171,58 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>true</td>
+  <td>
+    Whether to compute locality preferences for reduce tasks.
+  </td>
+  <td>1.5.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.mapOutput.minSizeForBroadcast</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1183]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1541-L1546

File: docs/configuration.md [hunk 13]
```diff
@@ -1063,6 +1171,58 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>false</td>
+  <td>
+    If enabled, part of a compressed/encrypted stream will be de-compressed/de-crypted by using extra memory 
+    to detect early corruption. Any IOException thrown will cause the task to be retried once 
+    and if it fails again with same exception, then FetchFailedException will be thrown to retry previous stage.
+  </td>
+  <td>3.0.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.useOldFetchProtocol</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1209]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1602-L1609

File: docs/configuration.md [hunk 14]
```diff
@@ -1102,6 +1262,22 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>3.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.service.db.enabled</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1266]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L710-L716

File: docs/configuration.md [hunk 15]
```diff
@@ -1735,6 +1911,14 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>1.6.0</td>
 </tr>
+<tr>
+  <td><code>spark.storage.unrollMemoryThreshold</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1915]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L400-L405

File: docs/configuration.md [hunk 16]
```diff
@@ -1745,6 +1929,16 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>2.2.0</td>
 </tr>
+<tr>
+  <td><code>spark.storage.localDiskByExecutors.cacheSize</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:1933]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1620-L1629

File: docs/configuration.md [hunk 17]
```diff
@@ -1816,6 +2010,14 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>2.1.1</td>
 </tr>
+<tr>
+  <td><code>spark.broadcast.UDFCompressionThreshold</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2014]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1934-L1941

File: docs/configuration.md [hunk 18]
```diff
@@ -1891,6 +2093,24 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>1.0.0</td>
 </tr>
+<tr>
+  <td><code>spark.files.ignoreCorruptFiles</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2097]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1073-L1079

File: docs/configuration.md [hunk 19]
```diff
@@ -1891,6 +2093,24 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>false</td>
+  <td>
+    Whether to ignore corrupt files. If true, the Spark jobs will continue to run when encountering corrupted or 
+    non-existing files and contents that have been read will still be returned.
+  </td>
+  <td>2.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.files.ignoreMissingFiles</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2106]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L1081-L1086

File: docs/configuration.md [hunk 20]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>0.9.2</td>
 </tr>
+<tr>
+  <td><code>spark.storage.decommission.enabled</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2168]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L451-L456

File: docs/configuration.md [hunk 21]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>false</td>
+  <td>
+    Whether to decommission the block manager when decommissioning executor.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.storage.decommission.shuffleBlocks.enabled</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2176]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L458-L464

File: docs/configuration.md [hunk 22]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>true</td>
+  <td>
+    Whether to transfer shuffle blocks during block manager decommissioning. Requires a migratable shuffle resolver 
+    (like sort based shuffle).
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.storage.decommission.shuffleBlocks.maxThreads</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2185]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L466-L472

File: docs/configuration.md [hunk 23]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>8</td>
+  <td>
+    Maximum number of threads to use in migrating shuffle files.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.storage.decommission.rddBlocks.enabled</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2193]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L474-L479

File: docs/configuration.md [hunk 24]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>true</td>
+  <td>
+    Whether to transfer RDD blocks during block manager decommissioning.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.storage.decommission.fallbackStorage.path</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2201]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L502-L510

File: docs/configuration.md [hunk 25]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>(none)</td>
+  <td>
+    The location for fallback storage during block manager decommissioning. For example, <code>s3a://spark-storage/</code>. 
+    In case of empty, fallback storage is disabled. The storage should be managed by TTL because Spark will not clean it up.
+  </td>
+  <td>3.1.0</td>
+</tr>
+<tr>
+  <td><code>spark.storage.decommission.fallbackStorage.cleanUp</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2210]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L512-L517

File: docs/configuration.md [hunk 26]
```diff
@@ -1944,6 +2164,67 @@ Apart from these, the following properties are also available, and may be useful
...
+  <td>false</td>
+  <td>
+    If true, Spark cleans up its fallback storage data during shutting down.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.storage.decommission.shuffleBlocks.maxDiskSize</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2218]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L519-L528

File: docs/configuration.md [hunk 27]
```diff
@@ -2321,6 +2630,16 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>2.4.1</td>
 </tr>
+<tr>
+  <td><code>spark.standalone.submit.waitAppCompletion</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:2634]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L2197-L2204

File: docs/configuration.md [hunk 28]
```diff
@@ -3342,6 +3661,15 @@ Push-based shuffle helps improve the reliability and performance of spark shuffl
   </td>
   <td>3.2.0</td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.push.numPushThreads</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:3665]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L2301-L2308

File: docs/configuration.md [hunk 29]
```diff
@@ -3360,6 +3688,15 @@ Push-based shuffle helps improve the reliability and performance of spark shuffl
   </td>
   <td>3.2.0</td>
 </tr>
+<tr>
+  <td><code>spark.shuffle.push.merge.finalizeThreads</code></td>
```
Comment (dcoliversun) [inline docs/configuration.md:3692]: https://github.com/apache/spark/blob/22483167e20208e40e24abe6898b2102ddaf4fc9/core/src/main/scala/org/apache/spark/internal/config/package.scala#L2330-L2338

**Comment by dcoliversun** [comment]
cc @HyukjinKwon @dongjoon-hyun 
It would be nice if you have a time to review this PR :)

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for working on this, @dcoliversun .

File: docs/configuration.md (hunk 3 above)
Comment (dongjoon-hyun) [reply docs/configuration.md:353]: BTW, I guess we don't want to expose this, @dcoliversun . There is no good for users.

File: docs/configuration.md [hunk 30]
```diff
@@ -847,6 +911,14 @@ Apart from these, the following properties are also available, and may be useful
   </td>
   <td>2.1.0</td>
 </tr>
+<tr>
+  <td><code>spark.plugins</code></td>
```
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply docs/configuration.md:906]: Yes, I agree to have this. It seems that we did mention only at `monitoring.md`.

[the remark this answers is not in the document]
Comment (dcoliversun) [reply docs/configuration.md:906]: Yeah, I see `spark.plugins` already mentioned at `monitoring.md`

File: docs/configuration.md (hunk 3 above)
Comment (dcoliversun) [reply docs/configuration.md:353]: OK. Better to mark as internal configuration? If so, I will make new PR to solve it.

**Review by HyukjinKwon — APPROVED** [review]
Took a quick look, LGTM. Thanks for working on this.

**Comment by dcoliversun** [comment]
Thanks for your help @srowen @HyukjinKwon @dongjoon-hyun

**Pull request closed without merging**